### PR TITLE
Following-up on meta display from on an offset

### PR DESCRIPTION
**Pull request by devnexen**
this time we display all metadata types.

### CODE REVIEW

File: libr/anal/meta.c [hunk 1]
```diff
@@ -727,6 +727,34 @@ static int meta_print_item(void *user, const char *k, const char *v) {
 	return 1;
 }
 
+R_API void r_meta_list_offset(RAnal *a, ut64 addr, char input) {
+	RAnalMetaUserItem ui = { a, 0, 0, NULL, NULL, 0, NULL };
```
Comment (radare) [inline libr/anal/meta.c]: ?

File: libr/anal/meta.c [hunk 2]
```diff
@@ -727,6 +727,34 @@ static int meta_print_item(void *user, const char *k, const char *v) {
...
+	SdbList *ls = sdb_foreach_list (DB, true);
+	SdbListIter *lsi;
+	SdbKv *kv;
+
+	ls_foreach (ls, lsi, kv) {
```
Comment (radare) [inline libr/anal/meta.c]: this is  very slow. see how the meta is accessed in other functions here, to reduce the scope of iterations it shifts the offset and uses that as a key to resolve the list of values that match a range of offsets.

Comment (radare) [reply libr/anal/meta.c]: just do r-meta-find

Comment (radare) [reply libr/anal/meta.c]: uhm. chk this:
```

R_API RAnalMetaItem *r_meta_find_in(RAnal *a, ut64 at, int type, int where) {
        char *res = meta_inrange_get (a, at, 1);
```

File: libr/anal/meta.c [hunk 3]
```diff
@@ -747,6 +776,8 @@ R_API int r_meta_list_cb(RAnal *a, int type, int rad, SdbForeachCallback cb, voi
 	isFirst = true; // TODO: kill global
 	ls_foreach (ls, lsi, kv) {
 		if (type == R_META_TYPE_ANY || (strlen (sdbkv_key (kv)) > 5 && sdbkv_key (kv)[5] == type)) {
+			if (strstr(sdbkv_key(kv), "83fc4"))
```
Comment (XVilka) [inline libr/anal/meta.c]: This was forgotten to be removed?

Comment (devnexen) [reply libr/anal/meta.c]: Yes :-) ....

**Pull request merged**